src/libraries/System.Private.Xml/src/System/Xml/XmlConvert.cs

```diff
@@ -1664,5 +1664,242 @@ internal static Exception CreateInvalidNameCharException(string name, int index,
         {
             return CreateException(index == 0 ? SR.Xml_BadStartNameChar : SR.Xml_BadNameChar, XmlException.BuildCharExceptionArgs(name, index), exceptionType, 0, index + 1);
         }
+
+        internal static bool TryFormat(bool value, Span<char> destination, out int charsWritten)
+        {
+            if (value)
+            {
+                if (destination.Length < 4)
+                {
+                    charsWritten = -1;
+                    return false;
+                }
+
+                destination[0] = 't';
```

Comment:
would it make sense to store `"true"u8` somewhere and just call CopyTo here? (same question for `false`)

Reply:
This should be faster, but if you prefer, I can change it.

Reply:
@krwq Done. I hope this was what you expect :

```
internal static bool TryFormat(bool value, Span<char> destination, out int charsWritten)
{
    ReadOnlySpan<char> valueSpan = value ? "true" : "false";

    charsWritten = valueSpan.Length;
    return valueSpan.TryCopyTo(destination);
}
```